cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestWaterUserEndpointInput.java

```diff
@@ -0,0 +1,94 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+class TestWaterUserEndpointInput {
+    @Test
+    void testGetAllQuery() {
+        String office = "SPK";
+        String projectId = "SACRAMENTO";
+        MockHttpRequestBuilder mockHttpRequestBuilder = new MockHttpRequestBuilder();
+        GetAll input = WaterUserEndpointInput.getAll(office, projectId);
+        input.addInputParameters(mockHttpRequestBuilder);
+        assertEquals(office, input.getOfficeId());
+        assertEquals(projectId, input.getProjectId());
+        assertEquals(ACCEPT_HEADER_V1, mockHttpRequestBuilder.getQueryHeader(ACCEPT_QUERY_HEADER));
+    }
+
+    @Test
+    void testGetOneQuery() {
+        String office = "SPK";
+        String userId = "user";
+        String projectId = "SACRAMENTO";
+        MockHttpRequestBuilder mockHttpRequestBuilder = new MockHttpRequestBuilder();
+        WaterUserEndpointInput.GetOne input = WaterUserEndpointInput.getOne(office, userId, projectId);
+        input.addInputParameters(mockHttpRequestBuilder);
+        assertEquals(office, input.getOfficeId());
+        assertEquals(userId, input.waterUserId());
+        assertEquals(projectId, input.getProjectId());
+        assertEquals(ACCEPT_HEADER_V1, mockHttpRequestBuilder.getQueryHeader(ACCEPT_QUERY_HEADER));
+    }
+
+    @Test
+    void testPostQuery() throws IOException {
+        String collect = readJsonFile("radar/v1/json/water_user.json");
+        WaterUser waterUser = RadarObjectMapper.mapJsonToObject(collect, WaterUser.class);
+        MockHttpRequestBuilder mockHttpRequestBuilder = new MockHttpRequestBuilder();
+        WaterUserEndpointInput.Post input = WaterUserEndpointInput.post(waterUser);
+        input.failIfExists(false); // Default is true
+        input.addInputParameters(mockHttpRequestBuilder);
+        assertEquals(waterUser, input.getWaterUser());
+        assertFalse(input.getFailIfExists());
+        assertEquals(ACCEPT_HEADER_V1, mockHttpRequestBuilder.getQueryHeader(ACCEPT_QUERY_HEADER));
+    }
+
+    @Test
+    void testDeleteQuery() {
+        String office = "SPK";
+        String userId = "user";
+        String projectId = "SACRAMENTO";
+        MockHttpRequestBuilder mockHttpRequestBuilder = new MockHttpRequestBuilder();
+        WaterUserEndpointInput.Delete input = WaterUserEndpointInput.delete(office, userId, projectId, DeleteMethod.ALL);
+        input.addInputParameters(mockHttpRequestBuilder);
+        assertEquals(office, input.getOfficeId());
+        assertEquals(userId, input.waterUserId());
+        assertEquals(projectId, input.getProjectId());
+        assertEquals(DeleteMethod.ALL.toString(), mockHttpRequestBuilder.getQueryParameter(WaterUserEndpointInput.Delete.METHOD_QUERY_PARAMETER));
+        assertEquals(ACCEPT_HEADER_V1, mockHttpRequestBuilder.getQueryHeader(ACCEPT_QUERY_HEADER));
+    }
+
+    @Test
+    void testDeleteNulls() {
```

Comment:
Test other endpoints required components

Reply:
Added new tests for other endpoints